runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -97,6 +97,33 @@ local function tags_lsp_to_vim(diagnostic, client_id)
   return tags
 end
 
+---@param related_informations lsp.DiagnosticRelatedInformation[]
+---@param offset_encoding string
+---@return vim.DiagnosticRelatedInformation?
+local function related_information_lsp_to_vim(related_informations, offset_encoding)
+  return vim.tbl_filter(function(item) return item ~= nil end,
+    --- @param related_information lsp.DiagnosticRelatedInformation
+    --- @return vim.DiagnosticRelatedInformation
+    vim.tbl_map(function(related_information)
+      local fname = vim.uri_to_fname(related_information.location.uri)
+      local bufnr = vim.fn.bufadd(fname)
+      if not bufnr then
+        return
+      end
+      local buf_lines = get_buf_lines(bufnr)
+      local start = related_information.location.range.start
+      local _end = related_information.location.range['end']
+      --- @type vim.DiagnosticRelatedInformation
+      return {
+        lnum = start.line,
+        col = line_byte_from_position(buf_lines, start.line, start.character, offset_encoding),
+        end_lnum = _end.line,
+        end_col = line_byte_from_position(buf_lines, _end.line, _end.character, offset_encoding),
+        message = related_information.message,
+      }
+    end, related_informations))
```

Comment:
Chaining a map with a filter is a good use case for `vim.iter`. `vim.iter(related_informations):map()` gives you filter+map semantics combined (if the callback function returns `nil` then the item is filtered out).

Although, is the original filter for `item ~= nil` even necessary? Presumably `related_informations` is an array, so it should not have any `nil` items in it.

Reply:
The intention was to handle the `if not bufnr then return end` path - But thanks for the hint about using `vim.iter(..):map` (the rustacean in me would've expected this to be called `filter_map` though)